flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -618,8 +623,27 @@ public void alterPartitionColumnStatistics(ObjectPath tablePath, CatalogPartitio
 
   @Override
   public List<CatalogPartitionSpec> listPartitions(ObjectPath tablePath)
-      throws CatalogException {
-    throw new UnsupportedOperationException();
+      throws TableNotExistException, TableNotPartitionedException, CatalogException {
+    Table table = loadIcebergTable(tablePath);
+
+    if (table.spec().isUnpartitioned()) {
+      throw new TableNotPartitionedException(icebergCatalog.name(), tablePath);
+    }
+    Set<CatalogPartitionSpec> set = Sets.newHashSet();
```

Comment:
Nit: please add empty lines after control flow blocks like `if`, `try`, `while`, and `for`.